apps/auth/service.py

```diff
@@ -45,6 +47,14 @@ def set_auth_default(self, doc, user_id):
         doc['token'] = utils.get_random_string(40)
         del doc['password']
 
+    def check_user(self, user):
```

Comment:
imo this should be part of `on_create` and not being called from every implementation,
authenticate should just check authentication and return a user, no matter if such
user is enabled etc. so it wouldn't do any authorization, just authentication